sparkfe/design_of_fetool.md

```diff
@@ -0,0 +1,107 @@
+- Start Date: 2021-05-13
+- Target Major Version: 1.0
+- Reference Issues: https://github.com/4paradigm/SparkFE/issues/58 and https://github.com/4paradigm/SparkFE/issues/69
+- Implementation PR: 
+
+# Summary
+
+We should provide some tools with easy-to-use API like Python for users to meet the featue extraction requirements like creating FEDB DDL.
+
+
+# Basic example
+
+Users can use fetool command to create FEDB DDL of creating tables.
+
+```
+fetool gen_ddl sql.yaml
+```
```

Comment:
`sql.yaml` is not friendly for user

Reply:
Currently we need the information of table map(registered table and parquet path, e.g. t1: file:///tmp/train_parquet/) and the sql script. The designed yaml format may be like this.

```
tables:
    t1: file:///tmp/train_parquet/
sql_file: /tmp/train.sql
```

However, the format of yaml file and command-line parameters should be discussed in another topic.